EKF/vel_pos_fusion.cpp

```diff
@@ -119,7 +120,7 @@ void Ekf::fuseVelPosHeight()
 	}
 
 	if (_fuse_height) {
-		if (_control_status.flags.baro_hgt) {
+		if (_control_status.flags.baro_hgt && !_control_status.flags.gps && !_control_status.flags.ev_pos && !(_control_status.flags.rng_hgt && (_R_to_earth(2, 2) > 0.7071f))) {
```

Comment:
I would prefer to fix the logic  that causes flags.baro_hgt to not go false when using external vision. I will have a look at that and push a fix for you to test.